src/concurrency/sync.rs

```diff
@@ -149,13 +150,68 @@ struct FutexWaiter {
     bitset: u32,
 }
 
+declare_id!(InitOnceId);
+
+struct InitOnceWaiter<'mir, 'tcx> {
+    thread: ThreadId,
+    callback: Box<dyn MachineCallback<'mir, 'tcx> + 'tcx>,
+}
+
+impl<'mir, 'tcx> std::fmt::Debug for InitOnceWaiter<'mir, 'tcx> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("InitOnce")
+            .field("thread", &self.thread)
+            .field("callback", &"dyn MachineCallback")
+            .finish()
+    }
+}
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+/// The current status of a one time initialization.
+pub enum InitOnceStatus {
+    Uninitialized,
+    Begun,
+    Complete,
+}
+
+impl Default for InitOnceStatus {
```

Comment:
Looks like a great use case for the new `#[default]` derive attribute :)

Reply:
Oh, it looks like it's stable. I thought it was still unstable.